services/actions/job_emitter.go

```diff
@@ -135,7 +142,35 @@ func (r *jobStatusResolver) resolve() map[int64]actions_model.Status {
 			if allSucceed {
 				ret[id] = actions_model.StatusWaiting
 			} else {
-				ret[id] = actions_model.StatusSkipped
+				// If a job's "if" condition is "always()", the job should always run even if some of its dependencies did not succeed.
+				// See https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds
+				always := false
```

Comment:
If it only needs to check "always()" here, I think it could be simplified to:

```
expr := strings.TrimSpace(strings.TrimSuffix(strings.TrimPrefix(expr, "${{"), "}}"))
always := expr == "always()"
```

Then there is no hard dependency to the "parser", no tricky type casting.